src/main/java/org/testng/IInvokedMethodListener.java

```diff
@@ -6,7 +6,27 @@
  */
 public interface IInvokedMethodListener extends ITestNGListener {
```

Comment:
Same here. I dont think we should be throwing exceptions.We should instead be having empty implementations.